### PR TITLE
952 include configuration for git mode in libms

### PR DESCRIPTION
**Pull request by nichlaes**

### CODE REVIEW

**Comment by nichlaes** [comment]
@prasadtalasila I have added modules now. Was this what you meant?

**Comment by prasadtalasila** [comment]
Module separation seems ok. Please use `useFactory` inside Files module

File: servers/lib/src/files/services/files-service.factory.ts
Comment (prasadtalasila) [inline servers/lib/src/files/services/files-service.factory.ts:1]: This goes into `useFactory` method inside `files.module.ts`

**Comment by nichlaes** [comment]
@prasadtalasila Can you have a look again please?

**Comment by prasadtalasila** [comment]
@nichlaes please rebase to `feature/distributed-demo` branch

**Comment by nichlaes** [comment]
@prasadtalasila  should be good, when I merge, it says it's up to date

**Review by prasadtalasila — COMMENTED** [review]
@nichlaes Thanks for the update. Please see the comments.

File: servers/lib/src/files/files-service.factory.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+@Injectable()
+export default class FilesServiceFactory {
+  static create(
+    configService: ConfigService,
+    fileServices: Map<string, IFilesService>
+  ): IFilesService {
+    const mode = configService.get<string>('MODE');
+
+    const service = fileServices.get(mode);
```
Comment (prasadtalasila) [inline servers/lib/src/files/files-service.factory.ts:13]: nice

File: servers/lib/src/files/interfaces/files.service.interface.ts [hunk 2]
```diff
@@ -1,7 +1,9 @@
 import { Project } from 'src/types.js';
 
+export const FILE_SERVICE = 'FILE_SERVICE';
 // FileService interface
 export interface IFilesService {
   listDirectory(path: string): Promise<Project>;
   readFile(path: string): Promise<Project>;
+  getFileMode(): string;
```
Comment (prasadtalasila) [inline servers/lib/src/files/interfaces/files.service.interface.ts:8]: Can this be named as `getMode`?

File: servers/lib/src/files/local/local-files.service.ts [hunk 3]
```diff
@@ -9,6 +9,9 @@ import { IFilesService } from '../interfaces/files.service.interface.js';
 export default class LocalFilesService implements IFilesService {
   // eslint-disable-next-line no-useless-constructor, no-empty-function
   constructor(private configService: ConfigService) {}
+  getFileMode(): string {
+    return 'local';
```
Comment (prasadtalasila) [inline servers/lib/src/files/local/local-files.service.ts:13]: Can this string be made a constant and defined at the top of this file?

File: servers/lib/src/files/local/local-files.service.ts [hunk 4]
```diff
@@ -9,6 +9,9 @@ import { IFilesService } from '../interfaces/files.service.interface.js';
...
+  }
 
   async listDirectory(path: string): Promise<Project> {
     const dataPath = this.configService.get('LOCAL_PATH');
```
Comment (prasadtalasila) [inline servers/lib/src/files/local/local-files.service.ts:17]: The `dataPath` variable does not change during the execution of the microservice. Can this be moved to constructor? Are there any implications for testing if it is moved there? Same is the case with `git-files.service.ts`.

**Review by prasadtalasila — COMMENTED** [review]
@nichlaes Please see the comments

File: servers/lib/src/files/files.module.ts [hunk 5]
```diff
@@ -1,9 +1,36 @@
...
+    FilesResolver,
+    {
+      provide: FILE_SERVICE,
+      useFactory: (
+        configService: ConfigService,
+        localFilesService: LocalFilesService,
+        gitFilesService: GitFilesService,
+      ) => {
+        const fileServices = new Map<CONFIG_MODE, IFilesService>([
+          [localFilesService.getMode(), localFilesService],
```
Comment (prasadtalasila) [inline servers/lib/src/files/files.module.ts:27]: @nichlaes 
The `getMode()` function can be directly invoked in `files-service.factory.ts`. Passing an array

`const fileServices = Array<IFilesService>([localFilesService, gitFilesService],])`

is sufficient

**Pull request merged**